### PR TITLE
update npm-coding-style.md document to reflect the use of standard

### PR DESCRIPTION
**Pull request by luislobo**
update npm-coding-style.md document to reflect the use of standard

### CODE REVIEW

File: doc/misc/npm-coding-style.md [hunk 1]
```diff
@@ -1,190 +1,1371 @@
...
+==================================================
 
 ## DESCRIPTION
 
-npm's coding style is a bit unconventional.  It is not different for
-difference's sake, but rather a carefully crafted style that is
-designed to reduce visual clutter and make bugs more apparent.
+npm's coding style follows [standard](https://github.com/standard/standard) rules.
+In the past it followed a different coding style but for the sake of standarization,
+now it aligns with the rules set by `standard`.
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md:8]: Why is there a line break here?

File: doc/misc/npm-coding-style.md [hunk 2]
```diff
@@ -1,190 +1,1371 @@
...
+  ```js
+  if (42 === age) { }    // ✗ avoid
+  if (age === 42) { }    // ✓ ok
+  ```
+
+## Semicolons
 
-The callback should always be the last argument in the list.  Its first
-argument is the Error or null.
+* No semicolons. (see: [1](http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding), [2](http://inimino.org/%7Einimino/blog/javascript_semicolons), [3](https://www.youtube.com/watch?v=gsfbh17Ax9I))
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: I think you should keep the exceptions part from the old document.

File: doc/misc/npm-coding-style.md [hunk 3]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
   This is the only gotcha with omitting semicolons, and `standard` protects you from this potential issue.
 
   (The full list is: `[`, `(`, `` ` ``, `+`, `*`, `/`, `-`, `,`, `.`, but most of these will never appear at the start of a line in real code.)
+## Functions
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: You say `## Functions` twice

File: doc/misc/npm-coding-style.md [hunk 4]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+
+Be very careful never to ever ever throw anything.  It's worse than useless.
+Just send the error message back as the first argument to the callback.
+
+## Errors
+## Functions
+
+Use named functions.  They make stack traces a lot easier to read.
+
+## Callbacks, Sync/async Style
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: Same with `## Callbacks`

File: doc/misc/npm-coding-style.md [hunk 5]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+way, the fs and http and child process stuff all uses the same callback-passing
+methodology.
+
+The callback should always be the last argument in the list.  Its first
+argument is the Error or null.
+
+Be very careful never to ever ever throw anything.  It's worse than useless.
+Just send the error message back as the first argument to the callback.
+
+## Errors
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: Why is there no content here?

File: doc/misc/npm-coding-style.md [hunk 6]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+## Case, naming, etc.
+
+Use `lowerCamelCase` for multiword identifiers when they refer to objects,
+functions, methods, properties, or anything not specified in this section.
+
+Use `UpperCamelCase` for class names (things that you'd pass to "new").
+
+Use `all-lower-hyphen-css-case` for multiword filenames and config keys.
+
+Use named functions.  They make stack traces easier to follow.
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: Third time you said this.

File: doc/misc/npm-coding-style.md [hunk 7]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+
+Use `CAPS_SNAKE_CASE` for constants, things that should never change
+and are rarely used.
+
+Use a single uppercase letter for function names where the function
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: But you said functions shouldn't be anonymous.

File: doc/misc/npm-coding-style.md [hunk 8]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+When something is intentionally missing or removed, set it to `null`.
+
+Don't set things to `undefined`.  Reserve that value to mean "not yet
+set to anything."
+
+Boolean objects are forbidden.
+Always create a new Error object with your message.  Don't just return a
+string message to the callback.  Stack traces are handy.
+
+## Logging
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: This is in twice.

File: doc/misc/npm-coding-style.md [hunk 9]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+
+Please clean up logs when they are no longer helpful.  In particular,
+logging the same object over and over again is not helpful.  Logs should
+report what's happening so that it's easier to track down where a fault
+occurs.
+
+Use appropriate log levels.  See `npm-config(7)` and search for
+"loglevel".
+
+## Case, naming, etc.
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: So is this.

File: doc/misc/npm-coding-style.md [hunk 10]
```diff
@@ -1317,7 +1317,133 @@ The following is an extract of the aforementioned document.
...
+Use named functions.  They make stack traces easier to follow.
+
+Use `CAPS_SNAKE_CASE` for constants, things that should never change
+and are rarely used.
+
+Use a single uppercase letter for function names where the function
+would normally be anonymous, but needs to call itself recursively.  It
+makes it clear that it's a "throwaway" function.
+
+## null, undefined, false, 0
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: And this

File: doc/misc/npm-coding-style.md [hunk 11]
```diff
@@ -1365,7 +1491,70 @@ The following is an extract of the aforementioned document.
   var nums = [1, 2, 3]
   nums.forEach(bar)
   ```
+## Functions
```
Comment (legodude17) [inline doc/misc/npm-coding-style.md]: This is the same block as above

**Comment by zkat** [comment]
Hi! We're moving repos to https://github.com/npm/cli/pulls! See [our blog post about the migration to npm.community](https://blog.npmjs.org/post/175587538995/announcing-npmcommunity) for details. As such, we're closing all active PRs on this repo.

Could you please re-open this PR against `npm/cli` instead? We're still interested in this patch so we hope you will! We'll continue discussion over there once it's moved. 💚

**Pull request closed without merging**